sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/InMemoryBaseTable.scala

```diff
@@ -434,7 +433,9 @@ abstract class InMemoryBaseTable(
     protected var streamingWriter: StreamingWrite = StreamingAppend
 
     override def overwriteDynamicPartitions(): WriteBuilder = {
-      assert(writer == Append)
+      if (writer != Append) {
```

Comment:
What is the reason for changing the exception type?

Reply:
The reason is we want to use catalyst test jar to submit ConnectService, so we can use `InMemoryTableCatalog`. But if `scalatest` used by `InMemoryTableCatalog`. We should add `scalatest` jar into ConnectService too. So I remove it.

Reply:
make sense